pandas/core/internals/managers.py

```diff
@@ -1965,12 +1965,18 @@ def items_overlap_with_suffix(left, lsuffix, right, rsuffix):
 
         def lrenamer(x):
             if x in to_rename:
-                return '{x}{lsuffix}'.format(x=x, lsuffix=lsuffix)
+                if lsuffix is None:
+                    return x
+                else:
+                    return '{x}{lsuffix}'.format(x=x, lsuffix=lsuffix)
             return x
 
         def rrenamer(x):
```

Comment:
Could maybe get rid of the need for two nearly identical functions with `functools.partial` (already imported):
```python
def renamer(x, suffix):
    if x in to_rename and suffix is not None:
        return '{x}{suffix}'.format(x=x, suffix=suffix)
    return x

lrenamer = partial(renamer, suffix=lsuffix)
rrenamer = partial(renamer, suffix=rsuffix)
```
Certainly more concise but maybe a bit harder to parse on the first reading of the code?  How do people feel about this suggestion?

Reply:
like it!! didn't notice this `partial` function!! thx!! @jschendel